### PR TITLE
kool-untyped 3rd

### PR DESCRIPTION
**Pull request by daejunpark**

### CODE REVIEW

**Comment by daejunpark** [comment]
@grosu please review.

**Comment by daejunpark** [comment]
@grosu please review this.

**Comment by daejunpark** [comment]
@grosu I think that all of your comments are either fixed or tracked by separate issues that will take more time to fix. I want you to confirm it, so that I can move on to another language.

File: k-distribution/tutorial/2_languages/2_kool/1_untyped/kool-untyped.k [hunk 1]
```diff
@@ -581,7 +596,14 @@ from SIMPLE unchanged. */
...
   rule N...M |-> _ => .Map  when N >Int M
-  rule N...M |-> K => N |-> K (N +Int 1)...M |-> K  when N <=Int M
+  rule N...M |-> K => N |-> K `(N +Int 1)...M |-> K`  when N <=Int M
+  // TODO(KORE):
+  // The additional bracket is required for disambiguity.
+  // Otherwise, this rule is possible to be parsed as:
+  //   rule N...M |-> K => N |-> ( (K(N +Int 1))...M |-> K )  when N <=Int M
+  // since K has no sort casting, thus can be considered as a klabel
+  // Note that backtick is better than parenthesis here,
+  // because parenthesis doesn't eliminate possiblility of K being of a klabel.
```
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/1_untyped/kool-untyped.k:606]: Not sure about that comment.  It should be possible to infer that K cannot be a KLabel there.
Besides, I think we should separate map binding using a comma, but that's something we can do after Jan 18.

**Comment by grosu** [comment]
As I told @laurayuwen, we should stop doing more definitions until we have what @cos and I agreed yesterday in place and working, basically the trick that allows us to not add the -COMMON modules and basically to not have to change the module structure we had before at all.

Once that happens, we will have to apply minimal changes to the tutorial languages.

Also, in terms of what else to work on, @daejunpark , what is really really needed is to make sure that, once we make the transition, we will not need to look back at 3.6.  Otherwise, by definitions, we are _not_ ready to transit.  In addition to making the tutorial examples work, which I am 100% sure that we will before the deadline, we also need to make sure that we do not lose any of the nice error messages that we had in 3.6, which we've spent many tens of hours to design.  Some of them are unnecessary now, because we do stronger consistency checks on K modules, but many of them are probably still necessary.  Without them, users/students will stumble into errors that they have no idea what they mean (e.g., uncaught exceptions).  I would say that that is probably the best thing you can really help us with, especially while you travel in Korea.  How does this sound?  Thanks!

File: k-distribution/tutorial/2_languages/2_kool/1_untyped/kool-untyped.k (hunk 1 above)
Comment (daejunpark) [inline k-distribution/tutorial/2_languages/2_kool/1_untyped/kool-untyped.k:606]: Isn't KLabel a subsort of K? Why do you think the variable K cannot be of a KLabel?

BTW, using commas as map separators will fix this problem.

Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/1_untyped/kool-untyped.k:606]: Of course KLabel is not a subsort of K.  Check kore.k.

Comment (daejunpark) [inline k-distribution/tutorial/2_languages/2_kool/1_untyped/kool-untyped.k:606]: Oh, you're right. I fixed the todo comment in dc269d68c75f015373c06124c000d87d23479564, and reported this in issue #1935 as well.

**Pull request merged**